### DISCUSSION TITLE
How can I add help "command"

### DISCUSSION THREAD

**Discussion by ghost** [bashly-framework/bashly#295] [Q&A] [answered]
How can I add the `help` **command**, which works the same as the `--help` **option**? like `docker help` and `docker --help`

**Comment by DannyBen** [comment, marked as answer]
Basically, you just create the command as you normally would, and have its code call the internal bashly help functions.

There is an example for that:
https://github.com/DannyBen/bashly/tree/master/examples/help-command#readme

This is the important bit:

```ruby
# src/help_command.sh
command="${args[command]}"
long_usage=yes

if [[ -z "$command" ]]; then
  # No command argument, show the global help
  help_function=cli_usage
else
  # Show the help for the requested command
  help_function="cli_${command}_usage"
fi

# Call the help function if it exists
if [[ $(type -t "$help_function") ]]; then
  "$help_function"
else
  echo "No help available for this command"
  exit 1
fi
```

Since this is not the first time this was asked, I wonder if there should be a simpler way to do this.

**Reply by ghost** [reply]
thanks for responding, no it's simple enough I just wasn't able to find it, thanks again for this wonderful tool.

**Reply by DannyBen** [reply]
I decided to also add a standard library function for it, so in the next release, people can simply run:

```
$ bashly add help
```

Implemented in #301
